Remove gui1 codebase

MichaelMauderer opened the pull request:
### Pull Request Description

Removes the old GUI1 code base and reduces the Rust code footprint by removing unused code.

### Important Notes

Updates build scripts and reformats part of the codebase with the autoformatter. 

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

MichaelMauderer committed:
Remove old app directory.

MichaelMauderer committed:
Move pack crate out of ensogl crate.

This is required as this crate is depended upon by the cli tool and thus blocking removal of the ensogl crate.

MichaelMauderer committed:
Remove ensogl crate.

MichaelMauderer committed:
Remove unused crates.

MichaelMauderer committed:
Remove unused crates.

MichaelMauderer committed:
Remove unused crates.

MichaelMauderer committed:
Remove unused crates.

MichaelMauderer committed:
Remove unused crates.

MichaelMauderer committed:
Remove unused crates.

MichaelMauderer committed:
Remove unused crates.

MichaelMauderer committed:
Remove unused crates.

MichaelMauderer committed:
Remove unused deps.

MichaelMauderer committed:
Fix build script.

MichaelMauderer committed:
Fix missing config yaml.

MichaelMauderer committed:
Reduce pub items in prelude.

MichaelMauderer committed:
Remove unused code.

MichaelMauderer committed:
Lint.

MichaelMauderer committed:
Fmt.

mwu-tow reviewed (changes requested):
While not run as part of PR checks, CI still builds old IDE for releases.
Find `UploadIde` and remove it. Similarly, `DeployGui` job should go — it deploys the old GUI to the cloud.

MichaelMauderer committed:
Remove obsolete jobs from build script.

MichaelMauderer committed:
Remove more references to ensogl and unused code.

MichaelMauderer committed:
Update workflows.

MichaelMauderer committed:
Fmt.

MichaelMauderer committed:
Update references to `ensogl-runner`.

MichaelMauderer committed:
Fix JS time APi access.

MichaelMauderer committed:
Remove unused import.

MichaelMauderer committed:
Remove more ensogl references.

MichaelMauderer committed:
Fmt and path update.

MichaelMauderer committed:
Merge branch 'develop' into wip/MichaelMauderer/remove_gui1_codebase

# Conflicts:
#	.github/workflows/release.yml

MichaelMauderer committed:
Update workflows.

MichaelMauderer committed:
Fmt.

MichaelMauderer committed:
Fix js time access.

MichaelMauderer committed:
Revert "Remove unused crates."

This reverts commit 5733b93f

MichaelMauderer committed:
Restore broken test.

MichaelMauderer committed:
Remove commented code.

somebody1234 reviewed (approved):
approving `ide-desktop` changes. most of them seem to be pretty minor.

somebody1234 commented:
also we should consider removing the local dashboard (`app/ide-desktop/lib/content/`) in favor of embedding the `cloud.enso.org` dashboard (`app/ide-desktop/lib/dashboard/`) directly into the electron app (`app/ide-desktop/lib/content/`) by editing the build scripts (`bundle.ts`)

farmaazon commented:
> also we should consider removing the local dashboard (`app/ide-desktop/lib/content/`) in favor of embedding the `cloud.enso.org` dashboard (`app/ide-desktop/lib/dashboard/`) directly into the electron app (`app/ide-desktop/lib/content/`) by editing the build scripts (`bundle.ts`)

Let's make it a separate issue.

farmaazon commented on app/gui2/config.yaml:
```diff
@@ -1,30 +1,30 @@
 # The name of an object created in the global window scope. This object will contain functions
 # to control the IDE, like the `main` function, and also the configuration object.
-windowAppScopeName: "enso"
+windowAppScopeName: 'enso'
```
This was meant to be shared configuration between rust and js projects; but now we have only js, and gui2 does not even use this file. We should reconsider if we want to keep this file (not in this PR, but perhaps continuation of the task).

MichaelMauderer replied:
It's still used in some parts of the JS code base, so it should be removed once those are refactored.

farmaazon commented on .github/workflows/release.yml:
```diff
@@ -436,12 +273,6 @@ jobs:
-      - enso-build-ci-gen-upload-ide-macos-aarch64
-      - enso-build-ci-gen-upload-ide-macos-x86_64
-      - enso-build-ci-gen-upload-ide-windows-x86_64
       - enso-build-ci-gen-upload-ide2-linux-x86_64
```
Let's rename ide2/gui2 to ide/gui everywhere.

farmaazon replied:
Or, after thinking about it a bit, I think we may leave old names - renaming may take a bit of time, and we want to [move and rename gui2 later](8484)

farmaazon commented on lib/rust/enso-pack/Cargo.toml:
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "ensogl-pack"
+name = "enso-pack"
```
I think we don't need this crate anymore. The rust remnants are compiled by gui2's package.json script directly calling wasm-pack.

MichaelMauderer replied:
I'll double-check and prepare a follow-up PR.

farmaazon commented on Cargo.toml:
```diff
@@ -7,33 +7,26 @@ resolver = "2"
+    "build/intellij-run-config-gen",
+    "build/deprecated/rust-scripts",
+    "lib/rust/*",
+    "lib/rust/parser/doc-parser",
+    "lib/rust/parser/src/syntax/tree/visitor",
+    "lib/rust/parser/jni",
+    "lib/rust/parser/generate-java",
+    "lib/rust/parser/schema",
+    "lib/rust/parser/debug",
+    "lib/rust/enso-pack",
+    "tools/language-server/logstat",
+    "tools/language-server/wstest",
 ]
```
Technically, every path dependency of a workspace member is also a workspace member IIRC. So this list could be reduced just to `lib/rust*` perhaps? `build` directory could also have only one, main package listed.

MichaelMauderer replied:
Changing it as suggested does cause build issues, do I'd opt for leaving it as is.

farmaazon commented on Cargo.toml:
```diff
@@ -7,33 +7,26 @@ resolver = "2"
 ]
 # The default memebers are those we want to check and test by default.
-default-members = ["app/gui", "lib/rust/*"]
+default-members = ["lib/rust/*"]
```
The default members were specified to not run integration tests by default. Let's remove this now.

farmaazon commented on build/ci_utils/src/actions/artifacts.rs:
```diff
@@ -12,7 +12,6 @@ use serde::de::DeserializeOwned;
 use tempfile::tempdir;
 
 
-
```
Why do we remove this line?

MichaelMauderer replied:
This was introduced by running `./run fmt` so I'd opt to leave it in.

farmaazon commented on lib/rust/launcher-shims/README.md:
```diff
@@ -1,58 +0,0 @@
-# Launcher Shims
```
Are you sure Engine don't need this crate anymore?

MichaelMauderer replied:
Double-checked with Radek and will restore it. It's not really used currently, but also not deprecated.

farmaazon commented on lib/rust/parser/src/macros/expand.rs:
```diff
@@ -5,6 +5,9 @@ use crate::prelude::*;
 
 use crate::syntax;
 
+use std::collections::HashMap;
+use std::collections::HashSet;
```
Arent these already imported by prelude?

MichaelMauderer replied:
To reduce the size of Prelude (and eventually remove it) that was removed.

farmaazon replied:
Hmm, and do we aim for prelude _removal_? I assumed it should be just slimmed down. And the common imports are one of the basic role of it.

farmaazon commented on lib/rust/parser/src/lib.rs:
```diff
@@ -240,15 +238,15 @@ fn expression_to_statement(mut tree: syntax::Tree<'_>) -> syntax::Tree<'_> {
-                        }));
-                        return tree;
-                    }
+                            // If the LHS is a type, this is a (destructuring) assignment.
+                            let lhs = expression_to_pattern(mem::take(lhs));
+                            tree.variant = Box::new(Variant::Assignment(Assignment {
+                                pattern: lhs,
+                                equals: mem::take(opr),
+                                expr: mem::take(rhs),
+                            }));
+                            return tree;
+                        }
```
Bad formatting

farmaazon commented on lib/rust/shapely/macros/src/derive_entry_point.rs:
```diff
@@ -7,8 +7,6 @@ use crate::prelude::*;
 // ===================
 
 fn crate_name_to_base_name(name: &str) -> String {
-    let name = name.replace("debug-scene-", "");
-    let name = name.replace("ensogl-example-", "");
```
We no longer have special entry points, so I think this entire module could be removed.

farmaazon commented on lib/rust/types/src/unit.rs:
```diff
@@ -1,874 +1,940 @@
-//! to implement many impls because the type `Unit` would be defined in an external crate.
-//!
-//! ## Macro Naming Pattern
-//! Many macros defined in this module use a specific naming, for example
-//! `impl_UNIT_x_FIELD_to_UNIT`. You should interpret it as a macro defining a transformation
-//! similar to `fn<Unit,Field>(unit:Unit, field:Field) -> Unit`, where `Field` is a specific field
-//! of the `Unit` (in most cases units have single field).
+//! [`Duration`] together, or divide one [`Duration`] by a number or another [`Duration`] and get
+//! [`Duration`] or a number, respectfully. You are allowed to define any combination of operators
+//! and rules of how the result inference should be performed.
 
```
So, I guess we don't keep the deprecated module, but replace it with unit2, and just the GH does not understand it?

MichaelMauderer replied:
Correct, there are just a few bits left that are used in the parser.

farmaazon reviewed (approved):
I think it's ok to merge (if it does not break CI), but it needs follow-up PR (or issue) which will remove enso(gl)-pack.

MichaelMauderer committed:
Revert "Remove unused crates."

This reverts commit 4da8cb6dad76f1854d288e37bac54a5265583182.

MichaelMauderer committed:
Remove default members.

MichaelMauderer committed:
Remove derive entry point.

MichaelMauderer committed:
Merge branch 'develop' into wip/MichaelMauderer/remove_gui1_codebase

MichaelMauderer committed:
Remove more things to fix linter.

mwu-tow committed:
[CI] Engine CI Rework, Part 1 (#9295)

I have created PR with the first set of changes for the Engine CI. The changes are small and effectively consist of:
1. Spltting the `verifyLicensePackages`. It is now run only on Linux. There are hardly any time benefits, as the actual job cost is dominated by the overhead of spinning a new job — but it is not expensive in the big picture.
2. Splitting the Scala Tests into separate job. This is probably the biggest "atomic" piece of work we have.
3. Splitting the Standard Library Tests into a separate job.

The time is nicely split across the jobs now. The last run has:
* 27 min for Scala tests;
* 25 min for Standard Library tests;
* 24 min for the "rest": the old job containing everything that has not been split.

While total CPU time has increased (as jobs are not effectively reusing the same build context), the wall time has decreased significantly. Previously we had ~1 hour of wall time for the old monolithic job, so we are getting more than 2x speedup.

The now-slowest Scala tests job is currently comparable with the native Rust tests (and they should improve when the old gui is gone) — which are the slowest job across all CI checks.

The PR is pretty minimal. Several future improvements can be made:
* Reorganizing and splitting other "heavy" jobs, like the native image generation.
* Reusing the built Engine distribution. However, this is probably a lower priority than I initially thought.
* Building package takes several minutes, so duplicating this job is not that expensive.
* The package is OS-specific.
* Scala tests don't really benefit from it, they'd need way more compilation artifacts.
It'd make sense to reuse the distribution if we, for example, decided to split more jobs that actually benefit from it, like Standard Library tests.
* Reusing the Rust build script binary.
* As our self-hosted runners reuse environment, we effectively get this for free. Especially when Rust part of codebase is less frequently changed.
* This is however significant cost for the GitHub-hosted runners, affecting our macOS runners. Reusing the binary does not save wall time for jobs that are run in parallel (as we have enough runners), but if we introduce job dependencies that'd force sequential execution of jobs on macOS, this would be a significant need.

MichaelMauderer committed:
Merge branch 'develop' into wip/MichaelMauderer/remove_gui1_codebase

# Conflicts:
#	app/gui/view/graph-editor/src/builtin/visualization/java_script/table.js
#	package-lock.json

MichaelMauderer committed:
Fix package-lock.json

MichaelMauderer committed:
Fix package-lock.json
